### PR TITLE
MAM-3089-friends-of-friends-dont-seem-to-be-running-correctly

### PR DESCRIPTION
**Pull request by jtomchak**
* Friends Of Friends list populating correctly
* Breadcrumb label showing only those smarlists that the user is subscribed too & enabled in FY
* Breadcrumbs are also purged when the user's feed is rebuilt.

### CODE REVIEW

File: app/controllers/api/v3/timelines/for_you_controller.rb [hunk 1]
```diff
@@ -109,7 +110,7 @@ def personalized_for_you_list_statuses
   end
 
   def personalzied_feed
-    ForYouFeed.new('personal', @account.id)
+    ForYouFeed.new('personal', @user[:acct])
   end
```
Comment (jtomchak) [inline app/controllers/api/v3/timelines/for_you_controller.rb:114]: account.id is tied to the moth.social instance. using the user's full username for cache key decouples it from having to do any moth.social specific lookup when read/writing to it. This is pervious logic from original FY

File: app/lib/mammoth/status_origin.rb [hunk 2]
```diff
@@ -64,6 +64,12 @@ def find(status_id, acct = nil)
         return results
     end 
 
+    # Delete All Status Origins by username
+    def reset(username)
+        list_key = key(username)
+        redis.keys("#{list_key}*").each { |key| redis.del(key) }
```
Comment (rileyhCode) [inline app/lib/mammoth/status_origin.rb:70]: Should we verify that list_key has a length (not empty) so we don't accidentally remove all entries?

**Review by rileyhCode — APPROVED** [review]
One comment for you to look at and make a change if you see fit. Then feel free to merge/push.

File: app/lib/mammoth/status_origin.rb (hunk 2 above)
Comment (jtomchak) [reply app/lib/mammoth/status_origin.rb:70]: oh. `* DEL` that's a good call. 😮

**Pull request merged**